mainwindow.cc

```diff
@@ -3241,11 +3241,11 @@ void MainWindow::setAutostart(bool autostart)
     }
     reg.sync();
 #else
-    // this is for KDE
+    // this is for Linux
     QString app_fname = QFileInfo(QCoreApplication::applicationFilePath()).baseName();
-    QString lnk(QDir::homePath()+"/.kde/Autostart/"+app_fname);
+    QString lnk(QDir::homePath()+"/.config/autostart/"+"goldendict.desktop");
     if (autostart) {
-        QFile f(QCoreApplication::applicationFilePath());
+        QFile f("/usr/share/applications/goldendict.desktop");
```

Comment:
The path to goldendict.desktop is not always like this. It depends on install prefix. For example, it is usually /usr/local/share/... if the package is installed bypassing system package manager.

Reply:
Yes, I use myself --prefix /usr with make.
Is it possible to use prefix variable here in Goldendict code?

Reply:
Try searching for the file with [QStandardPaths::locate](https://doc.qt.io/qt-5/qstandardpaths.html#locate)( QStandardPaths::ApplicationsLocation, "goldendict.desktop", QStandardPaths::LocateFile ).